### PR TITLE
Interstitial Page

### PR DESCRIPTION
**Pull request by danteruiz**
Interstitial Page (WIP)

- directions
1. Under developer menu check `Enable Interstitial`
2. Run the interstitialPage.js under system directory 

- note. The interstitial mode is disabled when physics is enabled. The is a current bug in master which physics does not get enabled. If this happens to you, toggle `Show Overlays` menu option under developer and the toolbar should appear, then just goto a different domain.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11815/HighFidelity-Beta-PR13744-048196e.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11815/HighFidelity-Beta-PR13744-048196e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11815/HighFidelity-Beta-Interface-PR13744-048196e.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11852/HighFidelity-Beta-PR13744-7622940.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11852/HighFidelity-Beta-PR13744-7622940.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11852/HighFidelity-Beta-Interface-PR13744-7622940.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11890/HighFidelity-Beta-PR13744-6a4a56c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11890/HighFidelity-Beta-PR13744-6a4a56c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11890/HighFidelity-Beta-Interface-PR13744-6a4a56c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11890/HighFidelity-Beta-PR13744-6a4a56c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11890/HighFidelity-Beta-Interface-PR13744-6a4a56c.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11926/HighFidelity-Beta-PR13744-a9fd034.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11926/HighFidelity-Beta-PR13744-a9fd034.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11926/HighFidelity-Beta-Interface-PR13744-a9fd034.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11926/HighFidelity-Beta-PR13744-a9fd034.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11926/HighFidelity-Beta-Interface-PR13744-a9fd034.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11962/HighFidelity-Beta-PR13744-0fed48b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11962/HighFidelity-Beta-PR13744-0fed48b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11962/HighFidelity-Beta-Interface-PR13744-0fed48b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11962/HighFidelity-Beta-PR13744-0fed48b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11962/HighFidelity-Beta-Interface-PR13744-0fed48b.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12005/HighFidelity-Beta-PR13744-353121b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12005/HighFidelity-Beta-PR13744-353121b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12005/HighFidelity-Beta-Interface-PR13744-353121b.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12031/HighFidelity-Beta-PR13744-46297c6.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12031/HighFidelity-Beta-PR13744-46297c6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12031/HighFidelity-Beta-Interface-PR13744-46297c6.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12031/HighFidelity-Beta-PR13744-46297c6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12031/HighFidelity-Beta-Interface-PR13744-46297c6.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12062/HighFidelity-Beta-PR13744-75cc1c6.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12062/HighFidelity-Beta-PR13744-75cc1c6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12062/HighFidelity-Beta-Interface-PR13744-75cc1c6.dmg

**Comment by danteruiz** [comment]
test this please

**Comment by danteruiz** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12241/HighFidelity-Beta-PR13744-b2fb2c7.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12241/HighFidelity-Beta-PR13744-b2fb2c7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12241/HighFidelity-Beta-Interface-PR13744-b2fb2c7.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12279/HighFidelity-Beta-PR13744-26a30ed.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12279/HighFidelity-Beta-PR13744-26a30ed.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12279/HighFidelity-Beta-Interface-PR13744-26a30ed.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12324/HighFidelity-Beta-PR13744-bb83a2f.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12324/HighFidelity-Beta-PR13744-bb83a2f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12324/HighFidelity-Beta-Interface-PR13744-bb83a2f.exe

File: libraries/entities-renderer/src/RenderableZoneEntityItem.cpp [hunk 1]
```diff
@@ -288,6 +288,16 @@ void ZoneEntityRenderer::doRenderUpdateSynchronousTyped(const ScenePointer& scen
         updateHazeFromEntity(entity);
     }
 
+
+    bool visuallyReady = true;
+    uint32_t skyboxMode = entity->getSkyboxMode();
+    if (skyboxMode == COMPONENT_MODE_ENABLED) {
+        bool skyboxLoadedOrFailed = (_skyboxTexture && (_skyboxTexture->isLoaded() || _skyboxTexture->isFailed()));
+        qDebug() << "------> " << skyboxLoadedOrFailed;
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderableZoneEntityItem.cpp]: did you mean to leave this in?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12403/HighFidelity-Beta-PR13744-af19849.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12403/HighFidelity-Beta-PR13744-af19849.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12403/HighFidelity-Beta-Interface-PR13744-af19849.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12408/HighFidelity-Beta-PR13744-92acaad.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12408/HighFidelity-Beta-Interface-PR13744-92acaad.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12408/HighFidelity-Beta-PR13744-92acaad.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12408/HighFidelity-Beta-Interface-PR13744-92acaad.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12412/HighFidelity-Beta-PR13744-e4293d9.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12412/HighFidelity-Beta-PR13744-e4293d9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12412/HighFidelity-Beta-Interface-PR13744-e4293d9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12412/HighFidelity-Beta-PR13744-e4293d9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12412/HighFidelity-Beta-Interface-PR13744-e4293d9.exe

File: libraries/entities-renderer/src/RenderableZoneEntityItem.cpp (hunk 1 above)
Comment (danteruiz) [reply libraries/entities-renderer/src/RenderableZoneEntityItem.cpp]: I was still debugging some things. I need to go through this PR to remove all my debug statements

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12418/HighFidelity-Beta-PR13744-e2cd695.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12418/HighFidelity-Beta-PR13744-e2cd695.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12418/HighFidelity-Beta-Interface-PR13744-e2cd695.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12418/HighFidelity-Beta-PR13744-e2cd695.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12418/HighFidelity-Beta-Interface-PR13744-e2cd695.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12531/HighFidelity-Beta-PR13744-93b0f15.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12531/HighFidelity-Beta-PR13744-93b0f15.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12531/HighFidelity-Beta-Interface-PR13744-93b0f15.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12531/HighFidelity-Beta-PR13744-93b0f15.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12531/HighFidelity-Beta-Interface-PR13744-93b0f15.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12533/HighFidelity-Beta-PR13744-03ede93.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12533/HighFidelity-Beta-PR13744-03ede93.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12533/HighFidelity-Beta-Interface-PR13744-03ede93.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12533/HighFidelity-Beta-PR13744-03ede93.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12533/HighFidelity-Beta-Interface-PR13744-03ede93.exe

**Comment by alexiamandeville** [comment]
Feedback:
Loading bar often looks like it's at 100% but still not loading the level. I'd say about 5/6 times.
Frequently I'm stuck in domain one once I've clicked on domain2 to load into. Then the loading screen comes up about 5 seconds later.
The overlays still look jittery, but that's an engine bug as I know it.
TheSpot button does not have a hover state.
I'm not totally happy with the interaction when you click the button to TheSpot because everything happens so fast. Will discuss with you and Mark to come up with solution.
The items are not totally center aligned (see attached image. Tip is not center aligned with the loading bar, and neither is the domain title.
image

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12713/HighFidelity-Beta-PR13744-ec2d89f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12713/HighFidelity-Beta-PR13744-ec2d89f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12713/HighFidelity-Beta-Interface-PR13744-ec2d89f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12713/HighFidelity-Beta-PR13744-ec2d89f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12713/HighFidelity-Beta-Interface-PR13744-ec2d89f.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12728/HighFidelity-Beta-PR13744-98dafa3.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12728/HighFidelity-Beta-PR13744-98dafa3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12728/HighFidelity-Beta-Interface-PR13744-98dafa3.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12728/HighFidelity-Beta-PR13744-98dafa3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12728/HighFidelity-Beta-Interface-PR13744-98dafa3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12748/HighFidelity-Beta-PR13744-2f91195.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12748/HighFidelity-Beta-Interface-PR13744-2f91195.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12748/HighFidelity-Beta-PR13744-2f91195.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12748/HighFidelity-Beta-Interface-PR13744-2f91195.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12756/HighFidelity-Beta-PR13744-45a99a9.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12756/HighFidelity-Beta-PR13744-45a99a9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12756/HighFidelity-Beta-Interface-PR13744-45a99a9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12756/HighFidelity-Beta-PR13744-45a99a9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12756/HighFidelity-Beta-Interface-PR13744-45a99a9.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
The above issues seem to be fixed. The only potential feedback I can give @danteruiz is when I am loading into The Spot, The Spot button is still on the Loading Page and clickable. Not sure if this is intentional or if its better to remove the button when loading The Spot.

File: libraries/audio-client/src/AudioClient.cpp [hunk 2]
```diff
@@ -664,11 +663,10 @@ void AudioClient::handleAudioEnvironmentDataPacket(QSharedPointer<ReceivedMessag
         _receivedAudioStream.setReverb(reverbTime, wetLevel);
     } else {
         _receivedAudioStream.clearReverb();
-   }
+    }
```
Comment (r3tk0n) [inline libraries/audio-client/src/AudioClient.cpp]: Might wanna remove this accidental space here.

Comment (danteruiz) [reply libraries/audio-client/src/AudioClient.cpp]: I'll fix that

**Comment by wayne-chen** [comment]
Merged everything into #13938.

**Pull request closed without merging**